nano.js

```diff
@@ -37,89 +37,86 @@ const mb = menubar({
+      // Disable Node.js integration
+      webPreferences.nodeIntegration = false
+    })
   })
-})
 
-mb.on('ready', () => {
-  const pluginHost = registerGlobalPluginHost()
-  const appManager = registerGlobalAppManager()
-
-  /* for testing:
-  appManager.launch({
-    name: 'grid-ui',
-    args: {
-      scope: {
-        component: 'terminal',
-        client: 'geth'
-      }
-    }
-  })
-  */
+  mb.on('ready', () => {
+    const pluginHost = registerGlobalPluginHost()
+    const appManager = registerGlobalAppManager()
 
-  // Unsure of linux distros behavior with menubar
-  // so for now we will always show on launch
-  // if (!startMinimized) {
-  //   mb.showWindow()
-  // }
-  mb.showWindow()
+    // Unsure of linux distros behavior with menubar
+    // so for now we will always show on launch
+    // if (!startMinimized) {
+    //   mb.showWindow()
+    // }
+    mb.showWindow()
 
-  /*
-  mb.window.webContents.openDevTools({
-    mode: 'detach'
+    mb.window.on('blur', function() {
+      // it prevents window from hiding if keepWindowOpen is checked on tray's context menu
+      !keepWindowOpen && mb.hideWindow()
+    })
   })
-  */
-  mb.window.on('blur', function() {
-    // it prevents window from hiding if keepWindowOpen is checked on tray's context menu
-    !keepWindowOpen && mb.hideWindow()
-  })
-})
 
-// right-click menu for tray
-mb.on('after-create-window', function() {
-  const contextMenu = Menu.buildFromTemplate([
-    {
-      label: 'Keep window open',
-      type: 'checkbox',
-      checked: keepWindowOpen,
-      click: () => {
-        keepWindowOpen = !keepWindowOpen
-      }
-    },
-    { type: 'separator' },
-    {
-      label: 'Feedback',
-      click: () => {
-        shell.openExternal(
-          'https://docs.google.com/forms/d/e/1FAIpQLSeJ4BtbvDVSnIFCKG6TmJo_tbSZql-NBZHes_-M6SyTDTjP0Q/viewform'
-        )
-        mb.hideWindow()
-      }
-    },
-    { type: 'separator' },
-    {
-      label: 'Quit',
-      click: () => {
-        mb.app.quit()
+  // right-click menu for tray
+  mb.on('after-create-window', function() {
+    const contextMenu = Menu.buildFromTemplate([
+      {
+        label: 'Keep window open',
+        type: 'checkbox',
+        checked: keepWindowOpen,
+        click: () => {
+          keepWindowOpen = !keepWindowOpen
+        }
+      },
+      { type: 'separator' },
+      {
+        label: 'Feedback',
+        click: () => {
+          shell.openExternal(
+            'https://docs.google.com/forms/d/e/1FAIpQLSeJ4BtbvDVSnIFCKG6TmJo_tbSZql-NBZHes_-M6SyTDTjP0Q/viewform'
+          )
+          mb.hideWindow()
+        }
+      },
+      { type: 'separator' },
+      {
+        label: 'Quit',
+        click: () => {
+          mb.app.quit()
+        }
       }
+    ])
+    mb.tray.on('right-click', () => {
+      mb.tray.popUpContextMenu(contextMenu)
+    })
+  })
+}
+
+const gotTheLock = mb.app.requestSingleInstanceLock()
+if (!gotTheLock) {
```

Comment:
I think this code block can use some additional documentation: a brief description and/or a link to documentation here.

Reply:
hard to see the changes because a lot of untouched code is recognized as diff.. was the formatting changed?

Reply:
yeah, most of the code is now on a `init()` function.